### PR TITLE
Discourse Addon

### PR DESCRIPTION
**Pull request by annando**
Discourse has got some mailing list mode, so that you can participate via mail client as well. We use this as a base for this addon. Although Discourse has a very mighty API, the write access is not enabled by default for users. So we are using the mailing list mode instead - but we are fetching many data via API.

There are currently several things that should be improved over time, but it seems to work here fine for the last days.

### CODE REVIEW

File: discourse/discourse.php [hunk 1]
```diff
@@ -0,0 +1,354 @@
+<?php
+
+/**
+ * Name: Discourse Mail Connector
+ * Description: Improves mails from Discourse in mailing list mode
+ * Version: 0.1
+ * Author: Michael Vogel <http://pirati.ca/profile/heluecht>
+ *
+ */
+//use DOMDocument;
```
Comment (MrPetovan) [inline discourse/discourse.php]: Please remove commented out code.

File: discourse/discourse.php [hunk 2]
```diff
@@ -0,0 +1,354 @@
...
+*/
+
+function discourse_install()
+{
+	Hook::register('email_getmessage',        __FILE__, 'discourse_email_getmessage');
+	Hook::register('connector_settings',      __FILE__, 'discourse_settings');
+	Hook::register('connector_settings_post', __FILE__, 'discourse_settings_post');
+}
+
+function discourse_uninstall()
```
Comment (MrPetovan) [inline discourse/discourse.php]: If your `*_uninstall` function doesn't have anything else than hook unregistration, you can scrap it entirely, the addon uninstalling process now removes all hooks related to this addon by default. This addresses obsolete hook registration as well without having to name them explicitly in `*_uninstall`.

File: discourse/discourse.php [hunk 3]
```diff
@@ -0,0 +1,354 @@
...
+function discourse_settings(App $a, &$s)
+{
+	if (!local_user()) {
+		return;
+	}
+
+	$enabled = intval(PConfig::get(local_user(), 'discourse', 'enabled'));
+	$css = ($enabled ? '' : '-disabled');
+
+	$s .= '<span id="settings_discourse_inflated" class="settings-block fakelink" style="display: block;" onclick="openClose(\'settings_discourse_expanded\'); openClose(\'settings_discourse_inflated\');">';
```
Comment (MrPetovan) [inline discourse/discourse.php]: Pease use an addon template instead. The goal is to have no HTML code in PHP files.

Comment (annando) [reply discourse/discourse.php]: I did it like Twitter does. Where can I have a look at?

File: discourse/README [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+can be fetched via Friendica. Then the user has to activate the
+mailing list mode in Discourse.
+
+The mailing list mode in Discourse knows two different options:
+1. Get all posts - including your own. This will create duplicates
+   if you post via Friendica.
+2. Don't get your own posts. Then you will missing all your posts
+   that you made directly on Discourse. Since you cannot create
+   a new post via this connector (only comments are possible)
+   this is not a goog choice either.
```
[the remark this answers is not in the document]
Comment (annando) [reply discourse/README]: goog loog!

File: discourse/discourse.php (hunk 3 above)
Comment (MrPetovan) [reply discourse/discourse.php]: https://github.com/friendica/friendica-addons/blob/b308f3bc0528b49a8ce0ccc9be21e047e6dff87d/advancedcontentfilter/advancedcontentfilter.php#L207-L235

Comment (MrPetovan) [reply discourse/discourse.php]: Even better: https://github.com/friendica/friendica-addons/blob/b308f3bc0528b49a8ce0ccc9be21e047e6dff87d/langfilter/langfilter.php#L43-L66

File: discourse/discourse.php [hunk 5]
```diff
@@ -45,25 +45,14 @@ function discourse_settings(App $a, &$s)
...
-	$s .= '</span>';
-
-	$field_checkbox = Renderer::getMarkupTemplate('field_checkbox.tpl');
-
-	$s .= Renderer::replaceMacros($field_checkbox, [
-		'$field' => ['enabled', L10n::t('Enable processing of Discourse mailing list mails'), $enabled, L10n::t('If enabled, incoming mails from Discourse will be improved so they look much better. To make it work, you have to configure the e-mail settings in Friendica. You also have to enable the mailing list mode in Discourse. Then you have to add the Discourse mail account as contact.')]
+	$t = Renderer::getMarkupTemplate('settings.tpl', 'addon/discourse/');
+	$s .= Renderer::replaceMacros($t, [
+		'$title'   => L10n::t('Discourse'),
+		'$css'     => ($enabled ? '' : '-disabled'),
```
Comment (MrPetovan) [inline discourse/discourse.php]: Please pass `$enabled` as a template variable and do the CSS class shenanigan in the template.

**Pull request merged**